File: Data Structures & Algorithms/string-encode-and-decode/submission-0.java
class Solution {

    public String encode(List<String> strs) {
        StringBuilder res = new StringBuilder();
        for ( String s : strs ) {
            res.append(s.length()).append('@').append(s);
        }
        return res.toString();
    }


    public List<String> decode(String str) {
        List<String> res = new ArrayList<>();
        int i = 0;

        while (i < str.length()) {
            // Find the position of '@'
            int j = i;
            while (j < str.length() && str.charAt(j) != '@') {
                j++;
            }

            // Extract the length of the next word
            int length = Integer.parseInt(str.substring(i, j));
            i = j + 1; // Move past '@'

            // Extract the word
            String word = str.substring(i, i + length);
            res.add(word);

            // Move `i` to the start of the next encoded word
            i += length;
        }
        return res;
    }
}

File: Data Structures & Algorithms/plus-one/submission-0.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        
        r = 0
        for i, n in enumerate(digits):
            r += n * 10**(len(digits) - i - 1)
        r += 1

        st = str(r)
        res = []
        
        for c in st:
            res.append(int(c))

        return res